src/neo/SmartContract/ApplicationEngine.Contract.cs

```diff
@@ -137,21 +149,21 @@ private void CallContractInternal(UInt160 contractHash, string method, Array arg
             if (md is null) throw new InvalidOperationException($"Method {method} Does Not Exist In Contract {contractHash}");
 
             ContractManifest currentManifest = Snapshot.Contracts.TryGet(CurrentScriptHash)?.Manifest;
-            if (currentManifest != null && !currentManifest.CanCall(contract.Manifest, method))
+            if (currentManifest != null && !currentManifest.CanCall(contractHash, contract.Manifest, method))
```

Comment:
Move `CanCall()` to `ContractState`? Or create an extension method?

Reply:
Done